sound/soc/sof/pcm.c

```diff
@@ -213,7 +247,14 @@ static int sof_pcm_hw_params(struct snd_soc_component *component,
 
 	dev_dbg(component->dev, "stream_tag %d", pcm.params.stream_tag);
 
-	/* send IPC to the DSP */
+	/* if this is a repeated hw_params without hw_free, skip setting up widgets */
+	if (!spcm->stream[substream->stream].list) {
+		ret = sof_pcm_setup_connected_widgets(sdev, rtd, spcm, substream->stream);
```

Comment:
I am still in the dark on how a dai/BE pipeline would be setup?

Reply:
The BE pipeline would be connected to the FE pipeline by DPCM and snd_soc_dapm_dai_get_connected_widgets() walks the DAPM graph starting from the FE and ends the walk at the BE. So the list it retrieves includes the BE pipeline too.